ddtrace/internal/packages.py

```diff
@@ -77,6 +79,42 @@ def get_distributions():
     return pkgs
 
 
+@callonce
+def get_package_distributions() -> t.Mapping[str, t.List[str]]:
+    """a mapping of importable package names to their distribution name(s)"""
+    try:
+        import importlib.metadata as importlib_metadata
+    except ImportError:
+        import importlib_metadata  # type: ignore[no-redef]
+
+    # Prefer the official API if available, otherwise fallback to the vendored version
+    if hasattr(importlib_metadata, "packages_distributions"):
+        return importlib_metadata.packages_distributions()
+    return _packages_distributions()
```

Comment:
Why not simply use this one and avoid all the above? Since it's going to be always available

Reply:
I am worried about keeping our vendored version up to date, if we have an official upstream version I'd rather use that.

so anyone using Python 3.8 + 3.9 will use this vendored version, and anyone on 3.7 using an older version of `importlib_metdata` (we don't specify a min required version, just a max).

I am not sure if this was the right decision, I considered just using the vendored version for everyone. If someone has a strong opinion, I don't have one.